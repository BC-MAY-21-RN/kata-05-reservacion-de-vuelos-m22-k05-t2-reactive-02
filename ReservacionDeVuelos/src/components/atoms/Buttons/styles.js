import {StyleSheet, Dimensions} from 'react-native';
import colors from '../../../consts/colors';

const styles = StyleSheet.create({
  touchable: {
    backgroundColor: colors.inactive,
    height: 40,
    width: Dimensions.get('screen').width - 60,
    marginLeft: 30,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  ortext: {
    alignSelf: 'center',
    marginVertical: '1.23%',
    color: colors.inactive,
    fontSize: 16,
  },
  icon: {position: 'absolute', left: 28},
  text: {color: colors.white, fontSize: 18},
  buttonActive: {backgroundColor: colors.bluetitle},
});

export default styles;
