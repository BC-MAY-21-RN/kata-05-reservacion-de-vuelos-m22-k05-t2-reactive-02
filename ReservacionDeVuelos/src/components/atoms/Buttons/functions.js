import {GoogleSignin} from '@react-native-google-signin/google-signin';
import auth from '@react-native-firebase/auth';
import styles from './styles';

async function onGoogleButtonPress() {
  const {idToken} = await GoogleSignin.signIn();
  const googleCredential = auth.GoogleAuthProvider.credential(idToken);
  return auth().signInWithCredential(googleCredential);
}

const createAccount = (email, password, userName) => {
  auth()
    .createUserWithEmailAndPassword(email, password)
    .then(() => {})
    .catch(error => {
      if (error.code === 'auth/email-already-in-use') {
      }
      if (error.code === 'auth/invalid-email') {
      }
    });
};

function validate(password, mail, userName, changeAlert) {
  const regexpassword =
    /^(?=(?:.*\d){1})(?=(?:.*[A-Z]){1})(?=(?:.*[a-z]){1})(?=(?:.*[@$?+*-¿!#%&/()=¡\-_]){1})\S{8,16}$/;
  const regexmail =
    /^(?:[^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*|"[^\n"]+")@(?:[^<>()[\].,;:\s@"]+\.)+[^<>()[\]\.,;:\s@"]{2,63}$/i;
  if (regexmail.test(mail) && regexpassword.test(password)) {
    createAccount(mail, password, userName);
  } else {
    changeAlert(true);
  }
}

const func = {
  onGoogleButtonPress,
  createAccount,
  validate,
  buttonstyle: buttonsActive => {
    return {
      ...styles.touchable,
      ...(buttonsActive ? styles.buttonActive : null),
    };
  },
};

export default func;
