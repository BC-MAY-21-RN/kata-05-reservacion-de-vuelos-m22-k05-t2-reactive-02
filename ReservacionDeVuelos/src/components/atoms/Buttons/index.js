import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import texts from '../../../consts/text';
import GoogleSVG from '../../../assets/icons/google.svg';
import styles from './styles';
import func from './functions';
import globalstyle from '../../../consts/globalstyles';

export default function Buttons({buttonsActive, form, changeAlert}) {
  return (
    <View style={globalstyle(15, 19, 30).marginTop}>
      <TouchableOpacity
        onPress={() => {
          func.validate(form.input3, form.input2, form.input1, changeAlert);
        }}
        style={func.buttonstyle(buttonsActive)}>
        <Text style={styles.text}>{texts.register.button1}</Text>
      </TouchableOpacity>
      <Text style={styles.ortext}>Or</Text>
      <TouchableOpacity
        onPress={() => func.onGoogleButtonPress().then(() => {})}
        style={func.buttonstyle(buttonsActive)}>
        <Text style={styles.text}>{texts.register.button2}</Text>
        <View style={styles.icon}>
          <GoogleSVG />
        </View>
      </TouchableOpacity>
    </View>
  );
}
