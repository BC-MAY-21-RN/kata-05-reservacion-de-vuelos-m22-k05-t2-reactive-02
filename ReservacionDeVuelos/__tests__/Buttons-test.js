import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import '@react-native-google-signin/google-signin';
import Buttons from '../src/components/atoms/Buttons';

jest.mock('@react-native-google-signin/google-signin', () => {});
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');

describe('Tests Buttons component', () => {
  it('Buttons snapshot', () => {
    const renderButtons = renderer
      .create(<Buttons buttonsActive={true} form={null} changeAlert={null} />)
      .toJSON();
    expect(renderButtons).toMatchSnapshot();
  });
});
