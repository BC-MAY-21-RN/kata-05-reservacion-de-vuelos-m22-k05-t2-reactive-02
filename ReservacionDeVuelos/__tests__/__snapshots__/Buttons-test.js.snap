// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Buttons component Buttons snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 30,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5C6DF8",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "marginLeft": 30,
        "opacity": 1,
        "width": 690,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#3C404855",
        "fontSize": 16,
        "marginVertical": "1.23%",
      }
    }
  >
    Or
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5C6DF8",
        "borderRadius": 10,
        "height": 40,
        "justifyContent": "center",
        "marginLeft": 30,
        "opacity": 1,
        "width": 690,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 18,
        }
      }
    >
      Sign Up with Google
    </Text>
    <View
      style={
        Object {
          "left": 28,
          "position": "absolute",
        }
      }
    >
      < />
    </View>
  </View>
</View>
`;
